metals/src/main/scala/scala/meta/internal/metals/BuildTargetClasses.scala

```diff
@@ -28,6 +28,27 @@ final class BuildTargetClasses(
     index.put(target, new Classes)
   }
 
+  def findMainClassByName(
+      name: String
+  ): List[(b.ScalaMainClass, b.BuildTargetIdentifier)] =
+    findClassesBy(_.mainClasses.values.find(_.getClassName() == name))
+
+  def findTestClassByName(
+      name: String
+  ): List[(String, b.BuildTargetIdentifier)] =
+    findClassesBy(_.testClasses.values.find(_ == name))
+
+  private def findClassesBy[A](
+      f: Classes => Option[A]
+  ): List[(A, b.BuildTargetIdentifier)] = {
```

Comment:
Would `Option` not be enough at least here?

Reply:
actually, I've made it a List motivated by your idea to report multiple source targets https://github.com/scalameta/metals/pull/1466#discussion_r384661635

Reply:
Ach, nvm. I just thought it's `toList` from Option. Nevermind then :sweat_smile: